system/security/shared/caching.cpp

```diff
@@ -549,7 +549,8 @@ inline void CPermissionsCache::removeAllManagedFileScopes()
 
     etc. Until full scope path checked, or no read permissions hit on ancestor scope.
 */
-static CriticalSection msCacheSyncCS;//for managed scopes cache syncronization
+static CriticalSection msCacheSyncCS;//for managed scopes cache synchronization
+static CriticalSection syncDefaultScopePermissions;//for cached default file scope permissions
```

Comment:
curios why this (and msCacheSyncCS) aren't members of CPermissionCache? (

Reply:
The protection for managed scopes was implemented that way, so I kept the same paradigm. I'd have to look deeper, but perhaps it is related to the option to share the cache across multiple instances of the security manager (as is the case with an ESP with multiple services loaded).